client/src/extension.ts

```diff
@@ -51,6 +51,27 @@ import { QUICKFIX_COMMAND, CoqWarningQuickFix } from './QuickFixProvider';
 let client: Client;
 
 export function activate(context: ExtensionContext) {
+
+    // Function to check for Coq files present in the workspace
+    function checkCoqFilesPresent() {
```

Comment:
I think you can replace this by the "workspaceContains" activation event in the package.json (see [here](https://code.visualstudio.com/api/references/activation-events#workspaceContains)). I am not sure if it requires workspace capabilities from the server that we still need to implement but I don't see why.

Reply:
I think it would be cool to use the activation event, the only downsides I can think of are:
- The `workspaceContains` is only emitted on folder open (not file creation or deletion). For example, this would make it so if we open an new/empty folder, then create a `_CoqProject` it won't make the activity bar entry appear 
  (this could maybe mitigated by adding `_CoqProject` to the recognized extensions for the coq language, although I think this would involve more separate work)

-  Similarly to above, since no deletion events are tracked the activity bar entry wont disappear if all `.v` and `_CoqProject` files disappear. This does seem like a corner case though, if someone ends up deleting all their Coq related files they can maybe deal with the activity bar entry sticking around.

Reply:
Okay ! I see your point ! I think it is fine as is then ! Could you just add ```_CoqProject``` as part of the ```checkCoqFilesPresent()``` routine ? I'll then merge !